File: Server/Domain/Entities/Author.cs
using Domain.Enums;

namespace Domain.Entities
{
    public class Author : BaseEntity
    {
        public Guid WorkId { get; set; }
        public Guid UserId { get; set; }
        public Guid? AuthorRoleId { get; set; }
        public Guid PurposeId { get; set; }
        public Guid? SCImagoFieldId { get; set; }
        public Guid? FieldId { get; set; }

        public int? Position { get; set; }
        public ScoreLevel? ScoreLevel { get; set; } // Mức điểm do người dùng chọn
        public decimal AuthorHour { get; set; }     // Giờ quy đổi mà tác giả nhận được
        public int WorkHour { get; set; }       // Giờ của công trình mà tác giả kê khai theo ScoreLevel
        public bool MarkedForScoring { get; set; }  // Cờ đánh dấu công trình được chọn để quy đổi
        public ProofStatus ProofStatus { get; set; } = ProofStatus.ChuaXuLy;
        public string? Note { get; set; }

        public virtual Field? Field { get; set; }
        public virtual SCImagoField? SCImagoField { get; set; }
        public virtual User? User { get; set; }
        public virtual AuthorRole? AuthorRole { get; set; }
        public virtual Work? Work { get; set; }
        public virtual Purpose? Purpose { get; set; }

        public virtual ICollection<WorkAuthor>? WorkAuthors { get; set; }
        public virtual AuthorRegistration? AuthorRegistration { get; set; }
    }
}

File: Client/src/lib/types/models/Author.ts
import { ProofStatus } from "../enums/ProofStatus";
import { AuthorRegistration } from "./AuthorRegistration";

export interface Author {
  id: string;
  workId: string;
  userId: string;
  authorRoleId: string;
  authorRoleName: string;
  purposeId: string;
  purposeName: string;
  scImagoFieldId?: string;
  scImagoFieldName?: string;
  fieldId?: string;
  fieldName?: string;
  position?: number;
  scoreLevel?: number;
  authorHour?: number;
  workHour?: number;
  proofStatus: ProofStatus;
  note?: string | null;
  authorRegistration?: AuthorRegistration | null;
  createdDate: string;
  modifiedDate?: string | null;
}

export interface UpdateAuthorRequest {
  authorRoleId?: string;
  purposeId?: string;
  position?: number;
  scoreLevel?: number;
  proofStatus?: ProofStatus;
  note?: string;
  scImagoFieldId?: string;
  fieldId?: string;
}

export interface CreateAuthorRequest {
  authorRoleId?: string;
  purposeId: string;
  position?: number;
  scoreLevel?: number;
  scImagoFieldId?: string;
  fieldId?: string;
}